.github/workflows/python-app.yml

```diff
@@ -105,15 +115,87 @@ jobs:
       run: |
         make lint-type-check
 
-    - name: Run Tests Coverage
+    - name: Run test on Node ${{ matrix.ci_node_index }}
       run: |
-        make coverage-report
+        set -e
+        mkdir -p test-results
+        rm -r test-results/parts || echo "no files to delete"
+        OUT=$(python -m tests --filepath test-results/results.xml --total-nodes ${{ matrix.ci_node_total }} --node-id ${{ matrix.ci_node_index }})
```

Comment:
where are you combining the result files?

Reply:
are you overwriting results.xml?

Reply:
I'm combining it in `coverage` workflow

Reply:
the result.xml will be overwritten only if we have any change in tests or the data